deploy/buildtools/buildconfig.py

```diff
@@ -66,8 +66,7 @@ def terminate_build_instance(self):
 
     def get_build_dir_name(self):
         """" Get the name of the local build directory. """
-        return """{}-{}-{}""".format(self.launch_time,
-                                     self.get_chisel_triplet(), self.name)
+        return """{}-{}""".format(self.launch_time, self.name)[:255]
```

Comment:
I don't think truncation is the play here. I'd prefer a warning. 

What exactly is struggling with the long filenames?

Reply:
If we do decide that these directories must be kept within 255 characters we should just throw an error on recipe names that are too long when they are pulled in. 

Also the directories under sim/generated-src + sim/output will suffer of this file length problem.

Reply:
Max filename size for most filesystems is 255 bytes.
I'll remove the truncation. Its actually pretty unlikely this will cause further errors with the builddir name if the chisel-triplet is no longer included.

The directories under sim/generated-src and sim/output will suffer the file length problem only if the chisel-triplet itself is > 255 chars.

Reply:
> The directories under sim/generated-src and sim/output will suffer the file length problem only if the chisel-triplet itself is > 255 chars.

Yeah i guess what i'm saying is that is all but assured if we continue using string concatenation + non-abbreviated class names.